ext/dom/node.c

```diff
@@ -1598,12 +1602,15 @@ static void dom_canonicalization(INTERNAL_FUNCTION_PARAMETERS, int mode) /* {{{
 		char *xquery;
 
 		tmp = zend_hash_str_find(ht, "query", sizeof("query")-1);
-		if (tmp && Z_TYPE_P(tmp) == IS_STRING) {
-			xquery = Z_STRVAL_P(tmp);
-		} else {
-			php_error_docref(NULL, E_WARNING, "'query' missing from xpath array or is not a string");
-			RETURN_FALSE;
+		if (!tmp) {
+			zend_argument_value_error(3, "\"query\" option must be in XPath array");
```

Comment:
```suggestion
			zend_argument_value_error(3, "must have a \"query\" key");
```
seems to be the standard message here.

Reply:
Also, another untested error condition :/